File: WinPaletter/Controllers/ComplexSave.vb
Imports WinPaletter.XenonCore

Public Class ComplexSave
    Private Sub XenonButton1_Click(sender As Object, e As EventArgs) Handles XenonButton1.Click
        Me.DialogResult = DialogResult.Yes
        Me.Close()
    End Sub

    Private Sub ComplexSave_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        Dim i1 As Integer = 0
        Dim i2 As Integer = 0

        If XenonRadioImage1.Checked Then i1 = 0
        If XenonRadioImage2.Checked Then i1 = 1
        If XenonRadioImage3.Checked Then i1 = 2

        If XenonRadioImage6.Checked Then i2 = 1
        If XenonRadioImage5.Checked Then i2 = 2
        If XenonRadioImage7.Checked Then i2 = 3
        If XenonRadioImage4.Checked Then i2 = 0

        My.Settings.General.ComplexSaveResult = i1 & "." & i2
        My.Settings.ThemeApplyingBehavior.ShowSaveConfirmation = XenonCheckBox2.Checked
        My.Settings.General.Save()
        My.Settings.ThemeApplyingBehavior.Save()
    End Sub

    Private Sub ComplexSave_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        LoadLanguage
        ApplyDarkMode(Me)

        Dim c As Color = PictureBox1.Image.AverageColor

        XenonAnimatedBox1.Color = c
        XenonAnimatedBox1.Color1 = c
        XenonAnimatedBox1.Color2 = c

        My.Computer.Audio.PlaySystemSound(Media.SystemSounds.Exclamation)

        Dim r As String() = My.Settings.General.ComplexSaveResult.Split(".")
        Dim r1 As String = r(0)
        Dim r2 As String = r(1)

        If My.W11 Then
            XenonRadioImage7.Image = My.Resources.Native11.Resize(20, 20)

        ElseIf My.W10 Then
            XenonRadioImage7.Image = My.Resources.Native10.Resize(20, 20)

        ElseIf My.W8 Then
            XenonRadioImage7.Image = My.Resources.Native8.Resize(20, 20)

        ElseIf My.W7 Then
            XenonRadioImage7.Image = My.Resources.Native7.Resize(20, 20)

        ElseIf My.WVista Then
            XenonRadioImage7.Image = My.Resources.NativeVista.Resize(20, 20)

        ElseIf My.WXP Then
            XenonRadioImage7.Image = My.Resources.NativeXP.Resize(20, 20)

        Else
            XenonRadioImage7.Image = My.Resources.Native11.Resize(20, 20)
        End If


        If r1 = 0 Then
            XenonRadioImage1.Checked = True
        ElseIf r1 = 1 Then
            XenonRadioImage2.Checked = True
        ElseIf r1 = 2 Then
            XenonRadioImage3.Checked = True
        Else
            XenonRadioImage3.Checked = True
        End If

        If r2 = 0 Then
            XenonRadioImage4.Checked = True
        ElseIf r2 = 1 Then
            XenonRadioImage6.Checked = True
        ElseIf r2 = 2 Then
            XenonRadioImage5.Checked = True
        ElseIf r2 = 3 Then
            XenonRadioImage7.Checked = True
        Else
            XenonRadioImage6.Checked = True
        End If

        XenonCheckBox2.Checked = My.Settings.ThemeApplyingBehavior.ShowSaveConfirmation

        Me.DialogResult = DialogResult.None
    End Sub

    Private Sub XenonButton2_Click(sender As Object, e As EventArgs) Handles XenonButton2.Click
        My.Settings.ThemeApplyingBehavior.ShowSaveConfirmation = XenonCheckBox2.Checked
        My.Settings.ThemeApplyingBehavior.Save()
        Me.DialogResult = DialogResult.No
        Me.Close()
    End Sub

    Private Sub XenonButton3_Click(sender As Object, e As EventArgs) Handles XenonButton3.Click
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub

    Public Function GetResponse(SaveFileDialog As System.Windows.Forms.SaveFileDialog, Apply_Theme_Sub As MethodInvoker, Apply_FirstTheme_Sub As MethodInvoker, Apply_DefaultWin_Sub As MethodInvoker) As Boolean
        If My.CP <> My.CP_Original AndAlso My.Settings.ThemeApplyingBehavior.ShowSaveConfirmation Then
            XenonGroupBox2.Enabled = Apply_Theme_Sub IsNot Nothing Or Apply_FirstTheme_Sub IsNot Nothing Or Apply_DefaultWin_Sub IsNot Nothing

            Select Case ShowDialog()
                Case DialogResult.Yes

                    Dim r As String() = My.Settings.General.ComplexSaveResult.Split(".")
                    Dim r1 As String = r(0)
                    Dim r2 As String = r(1)

                    Select Case r1
                        Case 0              '' Save
                            If IO.File.Exists(SaveFileDialog.FileName) Then
                                My.CP.Save(CP.CP_Type.File, SaveFileDialog.FileName)
                                My.CP_Original = My.CP.Clone
                            Else
                                If SaveFileDialog.ShowDialog = DialogResult.OK Then
                                    My.CP.Save(CP.CP_Type.File, SaveFileDialog.FileName)
                                    My.CP_Original = My.CP.Clone
                                Else
                                    Return False
                                End If
                            End If
                        Case 1              '' Save As
                            If SaveFileDialog.ShowDialog = DialogResult.OK Then
                                My.CP.Save(CP.CP_Type.File, SaveFileDialog.FileName)
                                My.CP_Original = My.CP.Clone
                            Else
                                Return False
                            End If
                    End Select

                    Select Case r2
                        Case 1
                            If Apply_Theme_Sub IsNot Nothing Then Apply_Theme_Sub()

                        Case 2
                            If Apply_FirstTheme_Sub IsNot Nothing Then Apply_FirstTheme_Sub()

                        Case 3
                            If Apply_DefaultWin_Sub IsNot Nothing Then Apply_DefaultWin_Sub()

                    End Select

                Case DialogResult.No

                Case DialogResult.Cancel
                    Return False
            End Select

            Return True
        Else
            Return False
        End If

    End Function
End Class

File: WinPaletter/Dialogs/ComplexSave.cs
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Windows.Forms;

namespace WinPaletter
{
    public partial class ComplexSave
    {
        public ComplexSave()
        {
            InitializeComponent();
        }
        private void Button1_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Yes;
            Close();
        }

        private void ComplexSave_FormClosing(object sender, FormClosingEventArgs e)
        {
            int i1 = 0;
            int i2 = 0;

            if (RadioImage1.Checked)
                i1 = 0;
            if (RadioImage2.Checked)
                i1 = 1;
            if (RadioImage3.Checked)
                i1 = 2;

            if (RadioImage6.Checked)
                i2 = 1;
            if (RadioImage5.Checked)
                i2 = 2;
            if (RadioImage7.Checked)
                i2 = 3;
            if (RadioImage4.Checked)
                i2 = 0;

            Program.Settings.General.ComplexSaveResult = i1 + "." + i2;
            Program.Settings.ThemeApplyingBehavior.ShowSaveConfirmation = CheckBox2.Checked;
            Program.Settings.General.Save();
            Program.Settings.ThemeApplyingBehavior.Save();
        }

        private void ComplexSave_Load(object sender, EventArgs e)
        {
            Icon = Forms.MainFrm.Icon;
            this.LoadLanguage();
            ApplyStyle(this);

            var c = PictureBox1.Image.AverageColor();

            AnimatedBox1.Color1 = c;
            AnimatedBox1.Color2 = c;

            Program.Computer.Audio.PlaySystemSound(System.Media.SystemSounds.Exclamation);

            string[] r = Program.Settings.General.ComplexSaveResult.Split('.');
            string r1 = r[0];
            string r2 = r[1];

            if (OS.W12)
            {
                RadioImage7.Image = Properties.Resources.Native12.Resize(20, 20);
            }

            if (OS.W11)
            {
                RadioImage7.Image = Properties.Resources.Native11.Resize(20, 20);
            }

            else if (OS.W10)
            {
                RadioImage7.Image = Properties.Resources.Native10.Resize(20, 20);
            }

            else if (OS.W8 | OS.W81)
            {
                RadioImage7.Image = Properties.Resources.Native8.Resize(20, 20);
            }

            else if (OS.W7)
            {
                RadioImage7.Image = Properties.Resources.Native7.Resize(20, 20);
            }

            else if (OS.WVista)
            {
                RadioImage7.Image = Properties.Resources.NativeVista.Resize(20, 20);
            }

            else if (OS.WXP)
            {
                RadioImage7.Image = Properties.Resources.NativeXP.Resize(20, 20);
            }

            else
            {
                RadioImage7.Image = Properties.Resources.Native11.Resize(20, 20);
            }


            if (Conversions.ToDouble(r1) == 0d)
            {
                RadioImage1.Checked = true;
            }
            else if (Conversions.ToDouble(r1) == 1d)
            {
                RadioImage2.Checked = true;
            }
            else if (Conversions.ToDouble(r1) == 2d)
            {
                RadioImage3.Checked = true;
            }
            else
            {
                RadioImage3.Checked = true;
            }

            if (Conversions.ToDouble(r2) == 0d)
            {
                RadioImage4.Checked = true;
            }
            else if (Conversions.ToDouble(r2) == 1d)
            {
                RadioImage6.Checked = true;
            }
            else if (Conversions.ToDouble(r2) == 2d)
            {
                RadioImage5.Checked = true;
            }
            else if (Conversions.ToDouble(r2) == 3d)
            {
                RadioImage7.Checked = true;
            }
            else
            {
                RadioImage6.Checked = true;
            }

            CheckBox2.Checked = Program.Settings.ThemeApplyingBehavior.ShowSaveConfirmation;

            DialogResult = DialogResult.None;
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            Program.Settings.ThemeApplyingBehavior.ShowSaveConfirmation = CheckBox2.Checked;
            Program.Settings.ThemeApplyingBehavior.Save();
            DialogResult = DialogResult.No;
            Close();
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        public bool GetResponse(SaveFileDialog SaveFileDialog, MethodInvoker Apply_Theme_Sub, MethodInvoker Apply_FirstTheme_Sub, MethodInvoker Apply_DefaultWin_Sub)
        {
            if ((Program.TM != Program.TM_Original) && Program.Settings.ThemeApplyingBehavior.ShowSaveConfirmation)
            {
                GroupBox2.Enabled = Apply_Theme_Sub is not null | Apply_FirstTheme_Sub is not null | Apply_DefaultWin_Sub is not null;

                switch (ShowDialog())
                {
                    case DialogResult.Yes:
                        {

                            string[] r = Program.Settings.General.ComplexSaveResult.Split('.');
                            string r1 = r[0];
                            string r2 = r[1];

                            switch (r1 ?? "")
                            {
                                case "0":              // ' Save
                                    {
                                        if (System.IO.File.Exists(SaveFileDialog.FileName))
                                        {
                                            Program.TM.Save(Theme.Manager.Source.File, SaveFileDialog.FileName);
                                            Program.TM_Original = (Theme.Manager)Program.TM.Clone();
                                        }
                                        else if (SaveFileDialog.ShowDialog() == DialogResult.OK)
                                        {
                                            Program.TM.Save(Theme.Manager.Source.File, SaveFileDialog.FileName);
                                            Program.TM_Original = (Theme.Manager)Program.TM.Clone();
                                        }
                                        else
                                        {
                                            return false;
                                        }

                                        break;
                                    }
                                case "1":              // ' Save As
                                    {
                                        if (SaveFileDialog.ShowDialog() == DialogResult.OK)
                                        {
                                            Program.TM.Save(Theme.Manager.Source.File, SaveFileDialog.FileName);
                                            Program.TM_Original = (Theme.Manager)Program.TM.Clone();
                                        }
                                        else
                                        {
                                            return false;
                                        }

                                        break;
                                    }
                            }

                            switch (r2 ?? "")
                            {
                                case "1":
                                    {
                                        if (Apply_Theme_Sub is not null)
                                            Apply_Theme_Sub();
                                        break;
                                    }

                                case "2":
                                    {
                                        if (Apply_FirstTheme_Sub is not null)
                                            Apply_FirstTheme_Sub();
                                        break;
                                    }

                                case "3":
                                    {
                                        if (Apply_DefaultWin_Sub is not null)
                                            Apply_DefaultWin_Sub();
                                        break;
                                    }

                            }

                            break;
                        }

                    case DialogResult.No:
                        {
                            break;
                        }

                    case DialogResult.Cancel:
                        {
                            return false;
                        }
                }

                return true;
            }
            else
            {
                return false;
            }

        }
    }
}